core/state.go

```diff
@@ -0,0 +1,39 @@
+package core

```

Comment:
small nit (which shouldn't be blocking): I find it a bit confusing that we have a `state.go` that contains an `Application`, as this is now the Application probably we should just name it `application.go`